src/writer/async_tokio.rs

```diff
@@ -82,6 +137,95 @@ impl<W: AsyncWrite + Unpin> Writer<W> {
     }
 }
 
+impl<'a, W: AsyncWrite + Unpin> ElementWriter<'a, W> {
+    /// Adds an attribute to this element.
+    pub fn with_attribute_async<'b, I>(mut self, attr: I) -> Self
+    where
+        I: Into<Attribute<'b>>,
+    {
+        self.start_tag.push_attribute(attr);
+        self
+    }
+
+    /// Add additional attributes to this element using an iterator.
+    ///
+    /// The yielded items must be convertible to [`Attribute`] using `Into`.
+    pub fn with_attributes_async<'b, I>(mut self, attributes: I) -> Self
+    where
+        I: IntoIterator,
+        I::Item: Into<Attribute<'b>>,
+    {
+        self.start_tag.extend_attributes(attributes);
+        self
+    }
```

Comment:
This methods does not depend from writer, so remove them. Move existing `with_attribute` and `with_attributes` methods to the `impl` block that does not restrict `W` type parameter, then they will work for both sync and async writers

Reply:
Since `with_attribute` is on the `ElementWriter`, the only way to do this is by removing the `Write` bound on the `ElementWriter` struct (in order to support a general impl. I still made the change.